### PR TITLE
fix a collection bug and simplify some code in collection

### PR DESCRIPTION
**Pull request by lgoettgens**

### CODE REVIEW

File: src/collection.jl [hunk 1]
```diff
@@ -253,6 +221,8 @@ function normalform(coll::Collector{T}, v::SyllableVector{T}) where T <: OscarIn
         end
         if b > 1
           push!(w, (y, b-1))
+        else
+          push!(w, (y, b+1))
```
Comment (ThomasBreuer) [inline src/collection.jl:225]: only if `b < -1` holds

**Review by ThomasBreuer — CHANGES_REQUESTED** [review]
Thanks for the improvements. Concerning the fix, we have to avoid zero as exponent.

File: src/collection.jl (hunk 1 above)
Comment (lgoettgens) [reply src/collection.jl:225]: no, we don't. we call `freely_reduced` enough times and that removes entries with zero exponent

Comment (ThomasBreuer) [reply src/collection.jl:225]: O.k., let us merge the fix.

**Pull request merged**